src/main/java/eu/deltasource/model/BankAccount.java

```
@@ -0,0 +1,220 @@
+        availableAmount -= amount;
+    }
+
+    public Owner getOwner() {
+        return owner;
+    }
+
+    public BankInstitution getBankInstitution() {
+        return bankInstitution;
+    }
+
+    public String getIban() {
+        return iban;
+    }
+
+    public String getCurrency() {
+        return currency;
+    }
+
+    private void setCurrency(String currency) {
+        if (currency.equals(Currency.BGN.getMessage()) || currency.equals(Currency.USD.getMessage()) || currency.equals(Currency.GBP.getMessage())) {
+            this.currency = currency;
+        } else {
+            throw new InvalidCurrencyException(ExceptionMessage.INVALID_CURRENCY.getMessage());
+        }
+    }
+
+    public double getAvailableAmount() {
+        return availableAmount;
+    }
+
+    public void setAvailableAmount(double availableAmount) {
+        if (availableAmount < 0) {
+            throw new AvailableAmountCannotBeNegativeException(ExceptionMessage.INVALID_AMOUNT.getMessage());
+        }
+        this.availableAmount = availableAmount;
+    }
+
+    public List<Transactions> getAccountTransactions() {
+        return Collections.unmodifiableList(accountTransactions);
+    }
+
+    public void addTransaction(Transactions transaction) {
+        accountTransactions.add(transaction);
+        getBankInstitution().addTransactions(transaction);
+    }
+
+    public List<AccountType> getAccountTypes() {
+        return Collections.unmodifiableList(accountTypes);
+    }
+
+    private void setAccountType(String accountType) {
+        if (accountType.equals(AccountType.CURRENT_ACCOUNT.getMessage())) {
+            accountTypes.add(AccountType.CURRENT_ACCOUNT);
+        } else if (accountType.equals(AccountType.SAVINGS_ACCOUNT.getMessage())) {
+            accountTypes.add(AccountType.SAVINGS_ACCOUNT);
+        } else {
+            throw new AccountTypeCannotBeDifferentFromCurrentAndSavingsException(ExceptionMessage.INVALID_ACCOUNT_TYPE.getMessage());
```

Same here, you can use InvalidInputException as in other places where you need to throw an exception for an invalid input.